### PR TITLE
MQTT, InfluxDB, Grafana for DHT and PIR

### PR DESCRIPTION
**Pull request by magley**
I had to completely refactor the device code because I couldn't get any work done otherwise.
There are instructions on how to implement checkpoint 2 features for other sensors in the README file.

Note: the only device I'm worried I "broke" when refactoring was the membrane keypad. If you have time, check to see if the logic is the same.

### CODE REVIEW

File: common/mqtt.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
+import json
+import paho.mqtt.publish as publish
+import threading
+
+
+class MqttSender:
+    def __init__(self, config: dict):
+        self.batch = []
+        self.counter = 0
```
Comment (elkcert) [inline common/mqtt.py]: Is this field redundant? `self.counter` is just `len(self.batch)`

File: components/mbkp.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+def read_real(output_pins, input_pins):
+    def _read_row(row: int, input_pins: list[int]):
+        GPIO.output(row.pin, GPIO.HIGH)
+        res = []
+        for idx, pin in enumerate(input_pins):
+            if GPIO.input(pin) == 1:
+                res.append(KEYPAD_CHARS[row.idx][idx])
+        GPIO.output(row.pin, GPIO.LOW)
+        return ''.join(res)
```
Comment (elkcert) [inline components/mbkp.py]: `row` is hinted at being an int (and is passed as int in below code), yet we access `row.idx` and `row.pin`. This crashes when `simulated` is set to `true` in the config.

**Review by elkcert — CHANGES_REQUESTED** [review]

File: common/mqtt.py (hunk 1 above)
Comment (magley) [reply common/mqtt.py]: [bd512b1](https://github.com/magley/my-pi-home/pull/6/commits/bd512b14ca2255faab587a060310bcdd08fc0f9d)

File: components/mbkp.py (hunk 2 above)
Comment (magley) [reply components/mbkp.py]: How about this? [ea7c694](https://github.com/magley/my-pi-home/pull/6/commits/ea7c69405199c00819eae523af59be41f70e5606)

**Review by elkcert — APPROVED** [review]

**Pull request merged**